xshop/users/serializers.py

```diff
@@ -3,33 +3,35 @@
 from django.contrib import auth
 from rest_framework.exceptions import AuthenticationFailed
 
+
 class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
-        fields = ('mobile', 'password' )
+        fields = ("mobile", "password")
+
 
 class LoginSerializer(serializers.ModelSerializer):
     mobile = serializers.CharField(max_length=11)
     password = serializers.CharField(max_length=25)
 
     class Meta:
         model = User
-        fields = ['mobile','password']
+        fields = ["mobile", "password"]
 
     def validate(self, attrs):
-        mobile = attrs.get('mobile', '')
-        passward = attrs.get('password', '')
+        mobile = attrs.get("mobile", "")
+        password = attrs.get("password", "")
 
         user = auth.authenticate(mobile=mobile, password=password)
         if not user.is_active:
-            raise AuthenticationFailed('Inactive account')
+            raise AuthenticationFailed("Inactive account")
         if not user.is_verified:
-            raise AuthenticationFailed('unverified account')
+            raise AuthenticationFailed("unverified account")
         if not user:
-            raise AuthenticationFailed('Invaled mobile or password, please try again')
-        
+            raise AuthenticationFailed("Invaled mobile or password, please try again")
```

Comment:
A serializer is used to validate the data, not to do further operations...

Here you should
1. validate user exists by filtering using the `mobile` field
2. check the password is correct for the incoming user
3. if there's an error, you should raise serializers.ValidationError
.. if there're any further operations other than data validation&serialization, those operations should be in the view